pinot-core/src/main/java/org/apache/pinot/core/query/optimizer/filter/MergeRangeFilterOptimizer.java

```diff
@@ -187,8 +189,88 @@ public Expression optimize(Expression filterExpression, @Nullable Schema schema)
         return filterExpression;
       }
     } else if (operator.equals(FilterKind.OR.name())) {
-      function.getOperands().replaceAll(c -> optimize(c, schema));
-      return filterExpression;
+      List<Expression> children = function.getOperands();
+      Map<String, Set<Range>> rangeMap = new HashMap<>();
```

Comment:
`Range` does not have `equals()` or `hashCode()` implemented, so should not be used in a `Set`